src/components/common/Comment.js

```diff
@@ -1,29 +1,105 @@
   const [newComment, setNewComment] = useState(""); // Store the new comment being typed by the user
+  const [hashtags, setHashtags] = useState([]); // Keeping track of available hashtags
+  const [selectedHashtags, setSelectedHashtags] = useState([]); // Keeping track of selected hashtags
+  const [checkedStatus, setCheckedStatus] = useState({});
+
+  useEffect(() => {
+    const fetchHashtags = async () => {
+      const Hashtags = Parse.Object.extend("Hashtags");
+      const query = new Parse.Query(Hashtags);
+
+      try {
+        const hashtagResults = await query.find();
+        setHashtags(hashtagResults);
+      } catch (error) {
+        console.error(`Error while fetching Hashtags: ${error}`);
+      }
+    };
+
+    // Call the fetch function
+    fetchHashtags();
+  }, []);
 
   // Function to add a new comment to the list
-  const addComment = () => {
+  const addComment = async () => {
+    // Fetch data of the current user
+    const currentUser = Parse.User.current();
+
     if (newComment) {
-      setComments([...comments, newComment]); // Update the list of comments with the new comment
-      setNewComment(""); // Clear the input field
+      try {
+        // Create Parse object for the Comment class and save it to the database
+        const Comment = Parse.Object.extend("Comment");
+        const commentToSave = new Comment();
+
+        // Set class attributes
+        commentToSave.set("CommentText", newComment);
+        commentToSave.set("UserID", currentUser);
+
+        // Set the LocationID for the comment
+        const location = new Parse.Object("Location");
+        location.id = locationId;
+        commentToSave.set("LocationID", location);
+
+        // Set the selected hashtags
+        const relation = commentToSave.relation("HashtagID");
+        selectedHashtags.forEach((hashtag) => relation.add(hashtag));
+
+        // Save the comment
+        const savedComment = await commentToSave.save();
+
+        // Comment object saved successfully
+        const commentObject = {
+          id: savedComment.id,
+          locationID: locationId,
+          text: newComment,
+          username: currentUser.get("username"),
+          experience: currentUser.get("experience"),
+          tags: selectedHashtags
+            .map((hashtag) => "#" + hashtag.get("Name"))
+            .join(" "), // Added tags to the comment
+        };
+
+        // Update the list of comments with the new comment
+        setComments([...comments, commentObject]);
+        // Clear the input field
+        setNewComment("");
+        // Clear selected hashtags
+        setSelectedHashtags([]);
+        // Reset checked status
+        setCheckedStatus({});
+
+        window.location.reload();
+      } catch (error) {
+        console.log(`Could not add comment. Error code: ${error}`);
+      }
+    }
+  };
+
+  // Call whenever Checkbox state changes
+  const handleCheckboxChange = (event, hashtagObj) => {
```

Comment:
this is a complicated function.
congrats.
looks good.

Reply:
💪  Thank you! it was tough to code